RecoTauTag/HLTProducers/src/TauTagFilter.cc

```diff
@@ -0,0 +1,156 @@
+/*
+ * \class TauTagFilter
+ *
+ * Filter tau candidates based on tagger scores.
+ *
+ * \author Konstantin Androsov, EPFL and ETHZ
+ */
+
+#include "DataFormats/BTauReco/interface/JetTag.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "HLTrigger/HLTcore/interface/HLTFilter.h"
+#include "RecoTauTag/RecoTau/interface/TauWPThreshold.h"
+
+class TauTagFilter : public HLTFilter {
+public:
+  using TauCollection = reco::PFJetCollection;
+  using TauTagCollection = reco::JetTagCollection;
+  using TauRef = reco::PFJetRef;
+  using Selector = tau::TauWPThreshold;
+  using LorentzVectorM = math::PtEtaPhiMLorentzVector;
+
+  explicit TauTagFilter(const edm::ParameterSet& cfg)
+      : HLTFilter(cfg),
+        nExpected_(cfg.getParameter<int>("nExpected")),
+        tausSrc_(cfg.getParameter<edm::InputTag>("taus")),
+        tausToken_(consumes<TauCollection>(tausSrc_)),
+        tauTagsToken_(consumes<TauTagCollection>(cfg.getParameter<edm::InputTag>("tauTags"))),
+        tauPtCorrToken_(mayConsume<TauTagCollection>(cfg.getParameter<edm::InputTag>("tauPtCorr"))),
+        seedsSrc_(mayConsume<trigger::TriggerFilterObjectWithRefs>(cfg.getParameter<edm::InputTag>("seeds"))),
+        seedTypes_(cfg.getParameter<std::vector<int>>("seedTypes")),
+        selector_(cfg.getParameter<std::string>("selection")),
+        minPt_(cfg.getParameter<double>("minPt")),
+        maxEta_(cfg.getParameter<double>("maxEta")),
+        usePtCorr_(cfg.getParameter<bool>("usePtCorr")),
+        matchWithSeeds_(cfg.getParameter<bool>("matchWithSeeds")),
+        matchingdR2_(std::pow(cfg.getParameter<double>("matchingdR"), 2)) {}
```

Comment:
A bit pedantic: in case of negative value of  `matchingR` one should switch off matching either by setting `matchWithSeeds` to `false` (batter) or `matchingR2` to negative value (worse). A warning should be printed out.